src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java

```diff
@@ -0,0 +1,92 @@
+package org.jenkinsci.plugins.ewm.casc;
+
+import jenkins.model.Jenkins;
+import org.jenkinsci.plugins.ewm.definitions.Disk;
+import org.jenkinsci.plugins.ewm.definitions.DiskPool;
+import org.jenkinsci.plugins.ewm.steps.ExwsAllocateStep;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.*;
+
+import org.jvnet.hudson.test.JenkinsRule;
+import io.jenkins.plugins.casc.ConfigurationAsCode;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Test for Configuration As Code Compatibility.
+ *
+ * @author Martin d'Anjou
+ */
+public class ConfigAsCodeTest {
+
+    @ClassRule public static JenkinsRule r = new JenkinsRule();
+
+    @Test
+    public void should_support_configuration_as_code() throws Exception {
+        String config = ConfigAsCodeTest.class.getResource("configuration-as-code.yaml").toString();
+        ConfigurationAsCode.get().configure(config);
+
+        // parse the yaml file and check if they match
+        File file  = new File(ConfigAsCodeTest.class.getResource("configuration-as-code.yaml").getFile());
+        String configContent = new String(Files.readAllBytes(file.toPath()));
+        Yaml yaml = new Yaml();
+        Map<String, Object> map = (Map<String, Object>) yaml.load(configContent);
+
+
+        // get the jenkins rule object.
+        Jenkins jenkins = r.getInstance();
+        ExwsAllocateStep.DescriptorImpl descriptor = (ExwsAllocateStep.DescriptorImpl) jenkins.getDescriptor(ExwsAllocateStep.class);
```

Comment:
```suggestion
        ExwsAllocateStep.DescriptorImpl descriptor = ExtensionList.lookupSingleton(ExwsAllocateStep.class);
```

Reply:
I changed it using ExtensionList.lookupSingleton but when i test it says it cannot find symbol "lookupSingleton".  Don't know what's happening...

Reply:
The jenkins.version of this plugin is too old,
needs something like 2.89 for that 😢 

It can be updated separately and more modern APIs could be used then

you should be able to use getDescriptorByType and not have to cast it I think

Reply:
This time would be OK now ! Thanks a lot !